File: C#/ContainsDuplicate.cs
//O(n)
//O(n)
public class Solution
{
    public bool ContainsDuplicate(int[] nums)
    {
        if (nums.Length == 0) return false;

        var hashSet = new HashSet<int>();

        for (int i = 0; i < nums.Length; i++)
        {
            if (hashSet.Contains(nums[i])) return true;
            else hashSet.Add(nums[i]);
        }

        return false;
    }
}

File: C++/containsDuplicate.cpp
// O(N)
// O(1)
#include <vector>
#include <set>

class Solution {
public:
    bool containsDuplicate(std::vector<int>& nums) {

        if(nums.empty()) return false;

        std::set<int> numbers;

        for(int i = 0; i < nums.size(); i++)
        {
            if(numbers.find(nums[i]) != numbers.end())
                return true;
            else
                numbers.insert(nums[i]);
        }

        return false;
    }
};